uaclient/contract.py

```diff
@@ -281,11 +322,27 @@ def _get_activity_info(self, machine_id: Optional[str] = None):
             if ent(self.cfg).user_facing_status()[0] == UserFacingStatus.ACTIVE
         ]
 
-        return {
-            "activityID": activity_id,
-            "activityToken": self.cfg.activity_token,
-            "resources": enabled_services,
-        }
+        activity_info = ActivityInfo(
```

Comment:
I think we can create the class only during the return call, just to make the code a little clearer.
Additionally, shouldn't we use `from_dict`  here to perform the type checks ?

Reply:
Good point!

If we construct it with the constructor, then type checking is handled by mypy.
`from_dict` gives us the type checking when we're creating it from a json or yaml file or other untyped dict source.